sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs

```diff
@@ -52,30 +43,24 @@ protected async Task PreProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
                 // As a result, before we know the auth scheme we need to avoid sending an unprotected body to Key Vault.
                 // We don't currently support this enhanced auth scheme in the SDK but we still don't want to send any unprotected data to vaults which require it.
 
-                RequestContent originalContent = message.Request.Content;
+                message.SetProperty(KeyVaultStashedContentKey, message.Request.Content);
                 message.Request.Content = null;
-
-                if (async)
-                {
-                    await ProcessNextAsync(message, pipeline).ConfigureAwait(false);
-                }
-                else
-                {
-                    ProcessNext(message, pipeline);
-                }
-
-                // set the content to the original content.
-                message.Request.Content = originalContent;
             }
             else
             {
                 // We fetched the scope from the cache, but we have not initialized the Scopes in the base yet.
-                Scopes = _scope.Scopes;
+                var context = new TokenRequestContext(_scope.Scopes, message.Request.ClientRequestId);
+                await AuthenticateRequestAsync(message, context, async).ConfigureAwait(false);
             }
```

Comment:
Isn't this handled above in line 31?
```c#
    if (_scope != null)
    {
        await AuthenticateRequestAsync(message, new TokenRequestContext(_scope.Scopes, message.Request.ClientRequestId), async).ConfigureAwait(false);
        return;
    }
```

Reply:
No - we get here in the case that the _scopeCache is populated from a previous instance of the policy, but a new instance of the policy has not yet populated _scope from the cache.